specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json

```diff
@@ -0,0 +1,1510 @@
+          "$ref": "#/definitions/CloudHsmClusterResource"
+        }
+      ],
+      "description": "Resource information with extended details.",
+      "type": "object"
+    },
+    "CloudHsmClusterResource": {
+      "properties": {
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentity"
+        },
+        "sku": {
+          "$ref": "#/definitions/CloudHsmClusterSku",
+          "description": "SKU details"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Cloud HSM Cluster Resource",
+      "type": "object",
+      "x-ms-azure-resource": true
+    },
+    "CloudHsmClusterProperties": {
+      "properties": {
+        "activationState": {
+          "type": "string",
+          "description": "State of security domain activation",
+          "readOnly": true,
+          "enum": [
+            "NotDefined",
+            "NotActivated",
+            "Active",
+            "Failed",
+            "Unknown"
+          ],
+          "x-ms-enum": {
+            "name": "ActivationState",
+            "modelAsString": true
+          }
+        },
+        "autoGeneratedDomainNameLabelScope": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster's auto-generated Domain Name Label Scope"
+        },
+        "fipsApprovedMode": {
+          "type": "boolean",
+          "default": "false",
+          "description": "False: Non-FIPS Mode with single-factor authentication. True: FIPS Approved Mode with single-factor authentication"
+        },
+        "hsms": {
+          "type": "array",
+          "readOnly": true,
+          "description": "An array of Cloud HSM Cluster's HSMs",
+          "x-ms-identifiers": [
+            "fqdn"
+          ],
+          "items": {
+            "$ref": "#/definitions/CloudHsmProperties"
+          }
+        },
+        "privateEndpointConnections": {
+          "type": "array",
+          "readOnly": true,
+          "description": "List of private endpoint connection resources",
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster's provisioningState",
+          "readOnly": true,
+          "enum": [
+            "Provisioning",
+            "Succeeded",
+            "Failed",
+            "Deleting",
+            "Canceled"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "publicNetworkAccess": {
+          "type": "string",
+          "description": "The Cloud HSM Cluster public network access"
```

Comment:
use enum. Please review the whole PR and check if objects are better modelled as enums rather than free form strings

Reply:
Updated string properties to Enums where applicable.